src/main/java/gyro/google/compute/DiskResource.java

```diff
@@ -0,0 +1,288 @@
+     *
+     * @subresource gyro.google.compute.EncryptionKey
+     */
+    @ConflictsWith({"source-disk-encryption-key", "source-snapshot-encryption-key"})
+    public EncryptionKey getSourceImageEncryptionKey() {
+        return sourceImageEncryptionKey;
+    }
+
+    public void setSourceImageEncryptionKey(EncryptionKey sourceImageEncryptionKey) {
+        this.sourceImageEncryptionKey = sourceImageEncryptionKey;
+    }
+
+    /**
+     * The unique ID of the image used to create the disk.
+     */
+    @Output
+    public String getSourceImageId() {
+        return sourceImageId;
+    }
+
+    public void setSourceImageId(String sourceImageId) {
+        this.sourceImageId = sourceImageId;
+    }
+
+    @Override
+    public void setResourcePolicies(List<String> resourcePolicies) {
+        super.setResourcePolicies(resourcePolicies != null
+            ? resourcePolicies.stream()
+                .map(policy -> ComputeUtils.toResourcePolicyUrl(getProjectId(), policy, getZone().substring(0, getZone().lastIndexOf("-"))))
+                .collect(Collectors.toList())
+            : null);
+    }
+
+    @Override
+    public void setType(String type) {
+        super.setType(type != null ? ComputeUtils.toZoneDiskTypeUrl(getProjectId(), type, getZone()) : null);
+    }
+
+    @Override
+    public void copyMore(Disk disk) {
+        setZone(disk.getZone().substring(disk.getZone().lastIndexOf("/") + 1));
+        setSourceImage(disk.getSourceImage());
+        setSourceImageId(disk.getSourceImageId());
+        setType(disk.getType());
+        setResourcePolicies(disk.getResourcePolicies());
+
+        if (disk.getSourceImageEncryptionKey() != null) {
+            EncryptionKey sourceImageEncryption = newSubresource(EncryptionKey.class);
+            sourceImageEncryption.copyFrom(disk.getSourceImageEncryptionKey());
+            setSourceImageEncryptionKey(sourceImageEncryption);
+        }
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state, Disk disk) {
+        Compute client = createComputeClient();
+
+        disk.setZone(getZone());
+        disk.setSourceImage(getSourceImage());
+        disk.setSourceImageEncryptionKey(getSourceImageEncryptionKey() != null ? getSourceImageEncryptionKey().toCustomerEncryptionKey() : null);
+
+        try {
+            Compute.Disks.Insert insert = client.disks().insert(getProjectId(), getZone(), disk);
+            Operation operation = insert.execute();
+            Operation.Error error = waitForCompletion(client, operation);
+            if (error != null) {
+                throw new GyroException(error.toPrettyString());
+            }
+
+            refresh();
+        } catch (Exception ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public List<ValidationError> validateMore() {
+        return new ArrayList<>();
+    }
```

Comment:
Rather than making an abstract `validateMore()` method that all concrete class need to implement, the concrete class can call its own validate method, removing the need for an empty implementation like this.